examples/baggage/go.mod

```diff
@@ -0,0 +1,50 @@
+module github.com/honeycombio/honeycomb-opentelemetry-go/examples/baggage
+
+go 1.19
+
+require (
+	github.com/honeycombio/honeycomb-opentelemetry-go v0.4.2
```

Comment:
If we move this to be a separate module, we should add a replace at the bottom of this file to link to the root project file path otherwise we have to wait for a release before the test can pick it up.

something like this should work:
```go
replace github.com/honeycombio/honeycomb-opentelemetry-go => ../../
```